### ISSUE TITLE
split `init` command into `init` and `build`

### ISSUE THREAD

**Issue by ashleygwilliams** [rustwasm/wasm-pack#188]
we've been *packing* (pun intended) a ton of stuff into the `init` command. this has the benefit of being a one-stop shop, but as we've seen with the intro of the `--mode` flag it's a lot harder to customize, and for those who use wasm-pack as a build tool, a single command that does *everything* is not really what folks want. for 0.5.0 let's break out `init` into a `init` and `build` command.

strawman proposal:

- `init` becomes what `init --mode no-build` is now
- `build` is everything `--mode no-build` skips, build should have a `--no-installs` flag

thoughts?

**Comment by ashleygwilliams** [comment]
related: #185

**Comment by fitzgen** [comment]
SGTM.

Ideally, `--no-installs` will be unnecessary as well, once we can properly determine whether the dependency tools are available and up-to-date or not.

**Comment by csmoe** [comment]
Given the original `init` options,
```rust
pub struct Init {
    crate_path: String,
    scope: Option<String>,
    disable_dts: bool,
    target: String,
    debug: bool,
    crate_name: String,
}
```
#194 introduced the configuration: 
```rust
[package.metadata.wasm-pack.build.<build_mode>]
debug = true
features = ["console_error_panic_hook"]
rustc-opt-level = 0
wasm-opt = false
wasm-snip = false
wasm-bindgen = true
```
In the addition of `build` options above, should I move `init.debug` into `build` too?

**Comment by ashleygwilliams** [comment]
hey @csmoe sorry i missed this comment. let me get the split done (hopefully today or tomorrow) and then we can figure out exactly where things should be. it'll be easier once it's actually in place.

**Comment by ashleygwilliams** [comment]
closed by #216
